lighthouse-core/lib/emulation.js

```diff
@@ -62,34 +62,44 @@ const NO_CPU_THROTTLE_METRICS = {
   rate: 1,
 };
 
-/**
- * @param {Driver} driver
- * @return {Promise<void>}
- */
-async function enableNexus5X(driver) {
-  await Promise.all([
-    driver.sendCommand('Emulation.setDeviceMetricsOverride', NEXUS5X_EMULATION_METRICS),
-    // Network.enable must be called for UA overriding to work
-    driver.sendCommand('Network.enable'),
-    driver.sendCommand('Network.setUserAgentOverride', {userAgent: NEXUS5X_USERAGENT}),
-    driver.sendCommand('Emulation.setTouchEmulationEnabled', {enabled: true}),
-  ]);
-}
+const emulationParams = {
+  mobile: {
+    userAgent: NEXUS5X_USERAGENT,
+    metrics: NEXUS5X_EMULATION_METRICS,
+    touchEnabled: true,
+  },
+  desktop: {
+    userAgent: DESKTOP_USERAGENT,
+    metrics: DESKTOP_EMULATION_METRICS,
+    touchEnabled: false,
+  },
+};
 
 /**
+ *
  * @param {Driver} driver
+ * @param {LH.Config.Settings} settings
  * @return {Promise<void>}
  */
-async function enableDesktop(driver) {
-  await Promise.all([
-    driver.sendCommand('Emulation.setDeviceMetricsOverride', DESKTOP_EMULATION_METRICS),
-    // Network.enable must be called for UA overriding to work
-    driver.sendCommand('Network.enable'),
-    driver.sendCommand('Network.setUserAgentOverride', {userAgent: DESKTOP_USERAGENT}),
-    driver.sendCommand('Emulation.setTouchEmulationEnabled', {enabled: false}),
-  ]);
+async function emulate(driver, settings) {
+  if (!settings.emulatedFormFactor || settings.emulatedFormFactor === 'none') return;
+  const params = emulationParams[settings.emulatedFormFactor];
+
+  // In DevTools, emulation is applied before Lighthouse starts (to deal with viewport emulation bugs)
+  // As a result, we don't double-apply viewport emulation.
```

Comment:
```suggestion
  // As a result, we don't double-apply viewport emulation (devtools sets `emulatedViewportMethod` to `provided`).
```

Reply:
otherwise I think it's confusing that the value is `devtools` but devtools doesn't use that option 😆